ark/mibi/tiling_utils.py

```diff
@@ -0,0 +1,240 @@
+import copy
+import datetime
+from itertools import combinations, product
+import json
+import os
+import random
+
+
+def set_tiling_params(fov_list_path, moly_path):
+    """Given a file specifying fov regions, set the MIBI tiling parameters
+
+    User inputs will be required for many values. Also returns moly_path data.
+
+    Args:
+        fov_list_path (str):
+            Path to the JSON file containing the fovs used to define each tiled region
+        moly_path (str):
+            Path to the JSON moly point file, needed to separate fovs
+
+    Returns:
+        tuple:
+            Contains:
+
+            - A dict containing the tiling parameters for each fov
+            - A dict defining the moly points to insert if specified
+    """
+
+    # file path validation
+    if not os.path.exists(fov_list_path):
+        raise FileNotFoundError("FOV region file %s does not exist" % fov_list_path)
+
+    if not os.path.exists(moly_path):
+        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+
+    # read in the fov list data
+    with open(fov_list_path, 'r') as flf:
+        fov_tile_info = json.load(flf)
+
+    # read in the moly point data
+    with open(moly_path, 'r') as mpf:
+        moly_point = json.load(mpf)
+
+    # define the parameter dict to return
+    tiling_params = {}
+
+    # retrieve the format version
+    tiling_params['fovFormatVersion'] = fov_tile_info['fovFormatVersion']
+
+    # define lists to hold the starting x and y coordinates for each region
+    region_start_x = []
+    region_start_y = []
+
+    # define lists to hold the number of fovs along each axis
+    fov_num_x = []
+    fov_num_y = []
+
+    # define lists to hold the size of each fov
+    x_fov_size = []
+    y_fov_size = []
+
+    # define a list to determine if the fovs should be randomly ordered
+    region_rand = []
+
+    # read in the data for each fov (region_start from fov_list_path, fov_num from user)
+    for fov in fov_tile_info['fovs']:
+        region_start_x.append(fov['centerPointMicrons']['x'])
+        region_start_y.append(fov['centerPointMicrons']['y'])
+
+        num_x = int(input("Enter number of x fovs for region %s: " % fov['name']))
+
+        while num_x < 1:
+            print("Error: number of fovs must be positive")
+            num_x = int(input("Enter number of x fovs for region %s: " % fov['name']))
```

Comment:
this can be condensed w/ do-while pattern
```
while True:
    num_x = int(input("..."))
    if num_x < 1:
        print("Error: ...")
```

Reply:
This do-while pattern will never exit regardless of what `num_x` is set to. You would need to add an `else: break` statement after the `if` statement for this to work, but I'm trying to avoid `break` statements as much as possible here.

Reply:
oops sorry I think I meant:
```
while True:
    num_x = int(input("..."))
    if num_x >= 1:
        break
    print("Error: ...")
```
regardless, since this 'force integer input' pattern is used a lot, it might be worth putting some version of it into its own separate function:
```
def int_input(msg, error_msg, condition):
    out = int(input(msg))
    if condition(out):
        return out
    print(error_msg)
```
which you could use like:
```
num_x = int_input('...', 'Error: ...', lambda x: x >= 1)
```